packages/sdk/src/waku/waku.ts

```diff
@@ -266,40 +264,46 @@ export class WakuNode implements IWaku {
   }
 
   public createDecoder(params: CreateDecoderParams): IDecoder<IDecodedMessage> {
-    const singleShardInfo = decoderParamsToShardInfo(
-      params,
-      this.networkConfig
+    const routingInfo = getRoutingInfo(
+      params.contentTopic,
+      this.networkConfig,
+      params.shardId
     );
-
-    log.info(
-      `Creating Decoder with input:${JSON.stringify(params.shardInfo)}, determined:${JSON.stringify(singleShardInfo)}, expected:${JSON.stringify(this.networkConfig)}.`
-    );
-
-    if (!isShardCompatible(singleShardInfo, this.networkConfig)) {
-      throw Error(`Cannot create decoder: incompatible shard configuration.`);
-    }
-
-    return createDecoder(params.contentTopic, singleShardInfo);
+    return createDecoder(params.contentTopic, routingInfo);
   }
 
   public createEncoder(params: CreateEncoderParams): IEncoder {
-    const singleShardInfo = decoderParamsToShardInfo(
-      params,
-      this.networkConfig
-    );
-
-    log.info(
-      `Creating Encoder with input:${JSON.stringify(params.shardInfo)}, determined:${JSON.stringify(singleShardInfo)}, expected:${JSON.stringify(this.networkConfig)}.`
+    const routingInfo = getRoutingInfo(
```

Comment:
similarly to my comment in `KeepAliveManager` - `getRoutingInfo` seems to belong `createEncoder`